src/main/java/net/minestom/server/inventory/click/ClickResult.java

```diff
@@ -0,0 +1,165 @@
+package net.minestom.server.inventory.click;
+
+import it.unimi.dsi.fastutil.ints.Int2ObjectArrayMap;
+import it.unimi.dsi.fastutil.ints.Int2ObjectFunction;
+import it.unimi.dsi.fastutil.ints.Int2ObjectMap;
+import it.unimi.dsi.fastutil.ints.Int2ObjectMaps;
+import net.minestom.server.entity.Player;
+import net.minestom.server.inventory.Inventory;
+import net.minestom.server.inventory.PlayerInventory;
+import net.minestom.server.item.ItemStack;
+import net.minestom.server.utils.inventory.PlayerInventoryUtils;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Map;
+
+/**
+ * Stores changes that occurred or will occur as the result of a click.
+ * @param player the player who clicked in the inventory
+ * @param clickedInventory the clicked inventory. This may be the player's inventory
+ * @param changes the map of changes that will occur to the inventory
+ * @param playerInventoryChanges the map of changes that will occur to the player inventory
+ * @param newCursorItem the player's cursor item after this click. Null indicates no change
+ * @param sideEffects the side effects of this click
+ */
+public record ClickResult(@NotNull Player player, @NotNull Inventory clickedInventory,
```

Comment:
Does the result need player + click inventory? Isn't it implicit from methods providing it?